src/jit/codegenxarch.cpp

```diff
@@ -6474,7 +6474,7 @@ void CodeGen::genJmpMethod(GenTreePtr jmp)
             }
         }
 
-#if FEATURE_VARARG
+#if FEATURE_VARARG && defined(_TARGET_AMD64_)
```

Comment:
This logic depends on the fact that in case of struct args, we don't promote the individual fields and register allocate those individual fields.  Please double check that assumption as well.